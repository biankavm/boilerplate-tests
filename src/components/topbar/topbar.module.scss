.nav {
  display: flex;
  gap: 16px;
  padding: 16px 24px;
  border-bottom: 1px solid #e0e0e0;
  background-color: #fff;
}

.navLink {
  padding: 8px 16px;
  text-decoration: none;
  color: #333;
  font-weight: 500;
  border-radius: 4px;
  transition: background-color 0.2s;

  &:hover {
    background-color: #f5f5f5;
  }
}
