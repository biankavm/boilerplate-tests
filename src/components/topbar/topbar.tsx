import Link from 'next/link'
import styles from './topbar.module.scss'

export function Topbar() {
  return (
    <nav className={styles.nav}>
      <Link href="/" className={styles.navLink}>
        Tarefas
      </Link>
      <Link href="/profile" className={styles.navLink}>
        Perfil
      </Link>
    </nav>
  )
}
