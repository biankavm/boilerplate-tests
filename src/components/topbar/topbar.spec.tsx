import { Topbar } from './topbar'
import { render, screen } from '@testing-library/react'

describe('Topbar', () => {
  const renderComp = () => render(<Topbar />)

  it('should render correctly', () => {
    const { container } = renderComp()
    expect(container).toMatchSnapshot()
  })

  it('should render text links', () => {
    renderComp()
    expect(screen.getByText(/tarefas/i)).toBeInTheDocument()
    expect(screen.getByRole('link', { name: /perfil/i }))
  })

  it('should links have correct attributes', async () => {
    renderComp()
    const links = screen.getAllByRole('link')
    expect(links[0]).toHaveAttribute('href', '/')
    expect(links[1]).toHaveAttribute('href', '/profile')
  })
})
