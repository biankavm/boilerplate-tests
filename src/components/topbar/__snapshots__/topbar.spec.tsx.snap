// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Topbar should render correctly 1`] = `
<div>
  <nav
    class="nav"
  >
    <a
      class="navLink"
      href="/"
    >
      Tarefas
    </a>
    <a
      class="navLink"
      href="/profile"
    >
      Perfil
    </a>
  </nav>
</div>
`;
